docs/sources/k6/next/misc/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,294 @@
+
+### Metrics
+
+We guarantee the stability of metric names, types, and units. Any changes to these are considered breaking and will be handled according to our [versioning strategy](#versioning-strategy). Doing so ensures that any product or artifact built on top of k6, such as dashboards, can maintain a reliable and consistent experience.
+
+## Deprecation, breaking changes, and migration policy
+
+### Breaking changes
+
+In a constantly evolving product, [breaking changes](#breaking-change) are sometimes unavoidable. They require that k6 users actively react to them because the state after the changes is not backward compatible with the previous state.
+
+Because we follow the [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html) versioning scheme, we work under the assumption breaking changes will always lead to a new major version. When the k6 team decides to introduce a breaking change, it will automatically define the need for a new major version.
+
+When k6 releases a new major version, we are expected to be explicit, clearly communicate the breaking changes, and provide instructions for migrating users.
+
+There are two types of breaking changes:
+
+1. Removal
+2. Update
+
+#### Steps to remove an API or a feature
+
+1. Deprecate a current API or a feature by following the dedicated policy.
+2. If necessary, add an alternative solution. Ideally, this would be an experimental version within a minor release within the current maintained state. Alternatively, it might be available in a Preview release.
+3. Provide documentation for the migration path.
+4. Proceed with removing the API and releasing a new major version.
+
+#### Steps to update an API or a feature
+
+1. Apply the changes, then publish them in the next major version. If the change is significant, it might be convenient to make it available in a Preview release that can support the migration path until a version is globally available.
+2. Provide documentation for the migration path.
+
+### Deprecation
+
+When the k6 team intends to remove an API or a feature, it will be marked as *deprecated*. This occurs when an API becomes obsolete, superseded by another API, or otherwise discontinued for business or technological reasons.
+Deprecated APIs remain available through their deprecated phase, which guarantees they will remain available at least throughout the entire life cycle of the current major version.
+
+A deprecation has several layers of communication necessary to be implemented and respected before it can be officially applied:
+
+1. Release notes and CHANGELOGs.
+2. Official documentation.
+3. When the user directly interacts with the feature.
+4. An alternative already available is recommended, but it shouldn’t be considered mandatory.
+
+### Migration
+
+When a breaking change is introduced, or an API is superseded, then documentation for a migration guide must be provided to users to help them understand how to move from version to version.
```

Comment:
another example of strange internal type language... I'd replace words like "must" with words like "will" or "are"